scripts/ui.js

```diff
@@ -19,26 +36,65 @@ function localizePrice (price) {
 }
 
 // Create pizza tiles list
-function createNewTile({id, title, price, image, ingredients}) {
+function createNewTile({id, title, price, image, ingredients}, i) {
     const ingredientsList = ingredients.join(", ");
     const fixedPrice = `${localizePrice(price)} zł`;
     const newTile = document.createElement('div');
 
     newTile.className = "tile"
     newTile.innerHTML = `
-        <h2 class="title">${id}. ${title}</h2>
+        <h2 class="title">${i + 1}. ${title}</h2>
         <div class="image">
             <img src="${image}" alt="${title}">
         </div>
-        <p class="ingredients"><span>Skladniki:</span><br>${ingredientsList}</p>
+        <p class="ingredients"><span>Skladniki:</span> ${ingredientsList}</p>
             <p class="price">${fixedPrice}</p>
             <button type="submit" class="btn-buy-product" data-name="${title}" data-price="${price}" data-id="${id}">Zamów</button>
     `
     containerList.appendChild(newTile)
 }
 
 function createNewList(list) {
-    list.forEach(item => createNewTile(item))
+    list.forEach((item, i) => createNewTile(item, i))
+}
+
+// Filtering by ingredients
+filterInput.addEventListener('keyup', (e) => filterTiles(e.target.value));
+
+function filterTiles(filterText = '', newData = data) {
+    const searchWords = filterText.toLowerCase().split(',').map(item => item.trim());
+    const filterData = newData.filter(item => searchWords.every(word=> {
+        return item.ingredients.join().toLowerCase().includes(word);
+    }));
+
+    if (filterData.length) {
+        tileList.innerText = '';
+        createNewList(filterData);
+    } else tileList.innerHTML = '<p class="filter-info">Nie mamy pizzy z takimi składnikami :(</p>';
+}
+
+// Pizza sorting
+sortOption.addEventListener('change', (e) => sortTiles(e.target.value));
+
+function sortTiles(sortOption) {
```

Comment:
funkcja powinna miec jedna odpowiedzialnosc, tej nazwa sugeruje ze ona sortuje, jednak nie, bo ona sortuje i filtruje, funkcja ktora filtruje, nie tylko filtruje, ale tez renderuje, powstaje z tego taki ciag wywolan i trzeba szukac co kogo wywoluje, lepiej jak jest to liniowe
sort()
filter()
render()

Reply:
Zmieniłem funkcje filtrowanie, sortowania i renderowania.

Reply:
funkcja `sortTiles` powinna:
1. posortować tablice bazując na wybranej opcji sortowania
2. wyrenderować na nowo posortowaną listę:

Reply:
@lukasz-galka @alexsimkovich 

obawiam się, że nie rozumiem :( Starałem się przebudować kod wg podpowiedzi Łukasza i (chyba) się udało. Wydaje mi się, że część komentarzy Alexa wskazują (tak przynajmniej to rozumiem) powrót do wersji wcześniejszej, gdzie Łukasz wskazał, żeby tak nie robić (ciąg wywołań). Jestem w kropce w związku z dzisiejszymi komentarzami od Alexa :(

Reply:
ok, inaczej. 
1. Funkcja `renderList` czemuś przyjmuje parametry do filtrowania i sortowania. Ta funkcja powinna tylko renderować liste bazując na tablice z pizzami
2. `sortTiles` jest ok, ale po sortowaniu, musisz wyrenderować posortowaną tablice (możesz zrobić to w samej funkcji **lub** osobno wywołać po wywołaniu funkcji `sortTiles` w `sortOption.addEventListener(...`, tylko wtedy `sortTiles` musi zwracać posortowaną tablice, którą przekażesz do renderowania
3.  `filterTiles` też jest ok i to samo jak z sortowaniem

Reply:
Dziękuję. Wieczorem jeszcze powalczę.

Reply:
@alexsimkovich jeszcze tylko podpytam. 

@lukasz-galka wyżej pisał wyżej _wiec powinienes miec jedna metode od renderowania listy i w niej robic sortowanie, fitrowanie i renderowanie_ (link: https://github.com/grumpyjacek/Intive_Patronage/pull/2#discussion_r764580221). I wydaje mi się, że tak jest.

Czy nie wrócimy do poprzedniej wersji, która już była, gdzie filtrowanie i sortowanie robiły też renderowanie? Przepraszam, że dopytuję, ale staram się ustalić kierunek działania.